File: biscuit/armv7a/measurements/toggle/linux/toggle.go
package main

import "os"

func main() {
	exporter, err := os.OpenFile("/sys/class/gpio/export", os.O_WRONLY,0777)
	if err != nil {
		panic(err)
	}
	exporter.WriteString("91")
	exporter.Close()

	direction, err := os.OpenFile("/sys/class/gpio/gpio91/direction", os.O_WRONLY,0777)
	if err != nil {
		panic(err)
	}
	direction.WriteString("out")
	direction.Close()

	value, err := os.OpenFile("/sys/class/gpio/gpio91/value", os.O_WRONLY, 0777)
	if err != nil {
		panic(err)
	}
	for i := 0; i < 1000000; i++ {
		value.WriteString("0")
		value.WriteString("1")
	}
	value.Close()
}

File: gert/armv7a/measurements/toggle/linux/toggle.c
#include <stdio.h>
#include <stdlib.h>
#include <fcntl.h>
#include <err.h>
#include <unistd.h>

int main()
{
  int exporter = open("/sys/class/gpio/export", O_WRONLY);
  if (exporter<0) {
    printf("fail to open exporter\n");
    return -1;
  }
  write(exporter, "91", 2);
  close(exporter);

  int direction = open("/sys/class/gpio/gpio91/direction", O_WRONLY);
  if (direction<0) {
    printf("fail to open direction\n");
    return -1;
  }
  write(direction, "out", 3);
  close(direction);

  int value = open("/sys/class/gpio/gpio91/value", O_WRONLY);
  if (value<0) {
    printf("fail to open value\n");
    return -1;
  }

  for (int i=0; i<1000000; i++)
  {
    write(value, "0", 1);
    write(value, "1", 1);
  }
  close(value);
  return 0;
}
